### PR TITLE
feat: image drag drop support in wysiwym

### PR DESCRIPTION
**Pull request by jpuri**
Implementing drag-drop for images.

For feature to work i demo pages we need to pass image upload fn as prop.

### CODE REVIEW

File: packages/react-tinacms-editor/src/state/plugins/Menu/Menu.tsx [hunk 1]
```diff
@@ -184,7 +176,7 @@ export const Menu = (props: Props) => {
               <ItalicControl view={view} />
               <UnderlineControl view={view} />
               <LinkControl view={view} />
-              <ImageMenu editorView={editorView} imageUpload={imageUpload} />
+              {/* <ImageMenu editorView={editorView} imageUpload={imageUpload} /> */}
```
Comment (jpuri) [inline packages/react-tinacms-editor/src/state/plugins/Menu/Menu.tsx]: As discussed over slack we can create a modal in editor for image upload and then enable this menu option.

File: packages/react-tinacms-editor/stories/wysiwyg.stories.tsx [hunk 2]
```diff
@@ -67,16 +67,19 @@ const WithImage = () => {
...
+          console.log(files)
+          return [
+            new Promise(resolve => {
+              setTimeout(function() {
+                resolve('https://i.imgur.com/2FCfbgg.jpg')
+              }, 250)
+            }),
+          ]
+        },
+        previewUrl: (str: string) => str,
```
Comment (jpuri) [inline packages/react-tinacms-editor/stories/wysiwyg.stories.tsx]: Storybook here demos way to use `ImageProps`.

File: packages/react-tinacms-editor/src/state/plugins/Image/index.ts [hunk 3]
```diff
@@ -18,55 +18,86 @@ limitations under the License.
...
+          const imagePromises = uploadImages(files)
+          imagePromises.forEach(promise => {
+            promise.then(url => {
+              const { state, dispatch } = editorView
+              insertImage(state, dispatch, url)
+            })
+          })
+        }
+        return false
+      },
```
Comment (jpuri) [inline packages/react-tinacms-editor/src/state/plugins/Image/index.ts]: `handleDrop` implemented to upload images.

File: packages/react-tinacms-editor/src/types.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+
+export interface ImageProps {
+  upload?: () => Promise<string>[]
```
Comment (ncphillips) [inline packages/react-tinacms-editor/src/types.ts]: I think this will end up being:

```
  upload?: () => Promise<string[]>
```

Comment (jpuri) [reply packages/react-tinacms-editor/src/types.ts]: Can you plz confirm, I can refactor code accordingly.

Comment (ncphillips) [reply packages/react-tinacms-editor/src/types.ts]: [Yup](https://github.com/tinacms/tinacms/blob/master/packages/@tinacms/media/src/media.ts#L59).

MediaStore returns array of Media objects. These will be passed as strings:

```js
upload: async (files) => {
  const media = await cms.media.store.persist(files)
  return media.map(toSrc)
}
```

Comment (jpuri) [reply packages/react-tinacms-editor/src/types.ts]: Hey @ncphillips , PR is updated.

File: packages/react-tinacms-editor/src/state/plugins/Image/index.ts (hunk 3 above)
Comment (jpuri) [reply packages/react-tinacms-editor/src/state/plugins/Image/index.ts]: I extended it to handle paste also.

**Comment by DirtyF** [comment]
I had to manually add `static/images` but now it works.

**Pull request merged**